### DISCUSSION TITLE
[BUG] ModuleNotFoundError: No module named 'emoji'

### DISCUSSION THREAD

**Discussion by madhatr** [Serene-Arc/bulk-downloader-for-reddit#943] [Q&A] [answered]
This is from FORK [bulk-downloader-for-reddit_with-saved-hashes](https://github.com/thomas694/bulk-downloader-for-reddit_with-saved-hashes)

I over wrote the bdfr folder with the bdfr folder inside the zip and this is the error I get. Before doing this bdfr works fine. 

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\user\AppData\Local\Programs\Python\Python312\Scripts\bdfr.exe\__main__.py", line 4, in <module>
  File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\site-packages\bdfr\__main__.py", line 11, in <module>
    from bdfr.archiver import Archiver
  File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\site-packages\bdfr\archiver.py", line 25, in <module>
    from bdfr.connector import RedditConnector
  File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\site-packages\bdfr\connector.py", line 32, in <module>
    from bdfr.file_name_formatter import FileNameFormatter
  File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\site-packages\bdfr\file_name_formatter.py", line 12, in <module>
    import emoji
ModuleNotFoundError: No module named 'emoji'
```

**Comment by Serene-Arc** [comment, marked as answer]
You need to install the module `emoji` from pip. Do this however your normally install python packages from pypi.

**Comment by SplitMilky** [comment]
So should this not be in the install instructions?  The fork says its exactly the same and does not mention extra requirements.

**Reply by Serene-Arc** [reply]
The OP did not follow the install instructions. They downloaded a zip file of the repository and extracted it. This is not what the instructions say to do. Doing that doesn't install any dependencies; installing through pip does.

**Reply by madhatr** [reply]
I did follow the instructions It says to copy the zip if not cloning with git. As I already had bdfr installed i chose to copy over as instructed.
![image](https://github.com/aliparlakci/bulk-downloader-for-reddit/assets/14855242/fb761c16-4b2e-463c-83e4-5e991085bc93)

The original does not use emoji, I assume it came from this version adding the ability for it.  
![image](https://github.com/aliparlakci/bulk-downloader-for-reddit/assets/14855242/ecef4eed-1110-426c-aa8b-029f137c5cbf)

Please tell me where I made the mistake.

**Reply by Serene-Arc** [reply]
That fork is not this one. Those instructions are not my instructions. You'll have to ask the developer of that repository.

**Reply by madhatr** [reply]
I know that is why my first sentence was this is for the FORK,  the fork has no issue or discussion page.  Wasn't sure where else to put it.

**Reply by Serene-Arc** [reply]
Not here. Forks that change or add functionality or have their own instructions are not relevant to this repository.
